examples/lock-app/linux/src/LockEndpoint.cpp

```diff
@@ -391,14 +404,32 @@ bool LockEndpoint::setLockState(DlLockState lockState, const Optional<chip::Byte
     }
 
     // Find the credential so we can make sure it is not absent right away
-    auto credential = std::find_if(mLockCredentials.begin(), mLockCredentials.end(), [&pin](const LockCredentialInfo & c) {
-        return (c.credentialType == CredentialTypeEnum::kPin && c.status != DlCredentialStatus::kAvailable) &&
-            chip::ByteSpan{ c.credentialData, c.credentialDataSize }.data_equal(pin.Value());
+    auto usercredentials = std::find_if(mLockCredentials.begin(), mLockCredentials.end(), [&pin](const auto & row) {
+        return std::find_if(row.begin(), row.end(), [&pin](const LockCredentialInfo & c) {
+                   return (c.status != DlCredentialStatus::kAvailable) &&
```

Comment:
This does not look right.  The incoming credential type here really is "pin", so we should only be looking at the PIN credentials, not all of them.  That should simplify this code a good bit...